test/jdk/java/foreign/TestSegmentAllocators.java

```diff
@@ -87,16 +85,14 @@ public <Z, L extends ValueLayout> void testAllocation(Z value, AllocationFactory
                     try {
                         allocationFunction.allocate(allocator, alignedLayout, value);
                         assertFalse(isBound);
-                    } catch (OutOfMemoryError ex) {
+                    } catch (IndexOutOfBoundsException | IllegalStateException ex) {
                         //failure is expected if bound
                         assertTrue(isBound);
```

Comment:
You are correct in the sense that slicing allocator returns NULL when it cannot allocate. However, if we allocate using e.g. `SegmentAllocator.allocate(ValueLayout.OfInt, int)` there will be issues because the returned segment has zero size and we will be trying to set a value on it. I think returning NULL here is perhaps too clever, we should just throw IOOBE on the allocate method - agree?

Reply:
Ah, I see. But, returning `NULL` seems nice for composing with other allocators. I guess you'd need to overwrite all the methods of SlicingAllocator to forward the `NULL` returned by the base `allocate`.

If you go for throwing an exception, I suggest sticking with OOME though.